### PR TITLE
Minor preparational JIT changes

### PR DESCRIPTION
**Pull request by izdeby**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21096 Minor preparational JIT changes**
* #21095 Updated scalar type to onnx mapping
* #21033 Enable all and any for bool tensors
* #21032 [WIP] Enabled add, sum and mul for bool tensor

This PR is a part of a stack which will change result tensor type of comparison ops from uint8 to bool. As this change is rather big and a lot of prep work is needed, im breaking it into a stack.

Changes in this PR:
- adding a bool scalar type check as `__ldg` is not supported for bool.

Im not quite sure how to test this, but you can take a look at the showcase [PR](https://github.com/pytorch/pytorch/pull/20831) where these changes are needed. 

Differential Revision: [D15546176](https://our.internmc.facebook.com/intern/diff/D15546176)

### CODE REVIEW

File: torch/csrc/jit/fuser/codegen.cpp [hunk 1]
```diff
@@ -402,14 +402,19 @@ std::string generateKernel(
...
+      const auto is_bool = input.second.has_value() && ((*input.second).scalar_type == at::ScalarType::Bool);
       if (is_half) {
         AT_ASSERT(use_cuda);
         env.s(
             "access",
             format("__half2float(t${formal}.data[t${formal}_offset])", env));
         has_half_tensor = true;
       } else if (use_cuda) {
-        env.s("access", format("__ldg(&t${formal}.data[t${formal}_offset])", env));
+        if (is_bool) {
```
Comment (ngimel) [inline torch/csrc/jit/fuser/codegen.cpp]: Comment here would be good (no __ldg for bools)

**Review by ngimel — APPROVED** [review]
Also, can you please add a test? Are any operations with bool tensors supposed to trigger fusion already?

**Comment by izdeby** [comment]
@ngimel, to be honest im not quite sure how to test this piece. There will be a lot of test exposure in the next PR where i will actually flip the switch and comparison ops will start returning bool tensor instead of uint8

**Comment by ngimel** [comment]
I think changes to codegen are either insufficient or unnecessary. You are changing load instruction, but it will be necessary only if there is a bool tensor argument to a kernel. Types of formal args are computed here:
https://github.com/pytorch/pytorch/blob/5fd99d94aa5b6921b05b7a18a11050af1837888e/torch/csrc/jit/fuser/codegen.cpp#L348, and `scalarTypeName` uses `FOR_ALL_SCALAR_TYPES_EXCEPT_HALF` macro https://github.com/pytorch/pytorch/blob/5fd99d94aa5b6921b05b7a18a11050af1837888e/torch/csrc/jit/fuser/codegen.cpp#L73. That macro does not define bool type https://github.com/pytorch/pytorch/blob/5fd99d94aa5b6921b05b7a18a11050af1837888e/c10/core/ScalarType.h#L102, so if there is indeed a bool tensor argument, codegen would break.

**Review by eellison — COMMENTED** [review]
I think you need to update scalarTypeFromJitType in jit_type.h so that Bool maps to the new bool tensor

**Comment by eellison** [comment]
Hi, sorry for the delay, do you mind changing https://github.com/pytorch/pytorch/blob/69e0f09d9922389b1586fd74b2828cb14ed03507/torch/csrc/jit/passes/shape_analysis.cpp#L947 to the new dtype ?

**Pull request closed without merging**